org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/util/BooleanGeneratorOption.java

```diff
@@ -0,0 +1,44 @@
+/*******************************************************************************
+ * Copyright (c) 2015, 2020 itemis AG (http://www.itemis.eu) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.xtext.xtext.generator.util;
+
+/**
+ * A Boolean-valued option with default value for use in generator fragments.
+ */
+public class BooleanGeneratorOption {
```

Comment:
Shouldn't this extend from GeneratorOption<T> or do have a separate class for the primitive returning `get`?

Reply:
good question. thought about this too but kept it for compatibility reasons